sale_order_type/models/sale.py

```diff
@@ -101,13 +114,25 @@ def create(self, vals):
     def write(self, vals):
         """A sale type could have a different order sequence, so we could
         need to change it accordingly"""
+        force_company = self.env.context.get("force_company")
+        if not force_company:
+            force_company = self.env.company.id
+        default_sequence = self._default_sequence_id()
         if vals.get("type_id"):
             sale_type = self.env["sale.order.type"].browse(vals["type_id"])
             if sale_type.sequence_id:
                 for record in self:
+                    # An order with a type without sequence would get the default one.
+                    # We want to avoid changing the order reference when the new
+                    # sequence has the same default sequence.
+                    ignore_default_sequence = (
+                        not record.type_id.sequence_id
+                        and sale_type.sequence_id == default_sequence
```

Comment:
should be sale_type.sequence_id.id == default_sequence.id?

Reply:
We don't need to. We're comparing at object level.